src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  isAlive() {
+    return this.health > 0;
+  }
 
+  checkDeath() {
+    if (!this.isAlive()) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
+  }
+}
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
```

Comment:
[FUNCTION NAMING] - if a method's name is `hide` - this method should just hide(set `true` to the object's `hidden` property) but not toggle the property's value. In this case, your `hide` method is correct as it sets `this.hidden` to `true`. Make sure it does not toggle the `hidden` property.